src/verify.c

```diff
@@ -1164,7 +1282,18 @@ enum swupd_code verify_main(void)
 	/* this concludes the critical section, after this point it's clean up time, the disk content is finished and final */
 
 clean_and_exit:
+	if (official_manifest) {
+		official_manifest->files = all_files;
```

Comment:
why do you need to recover official_manifest->files at the end of the function? Can't you free all_files when you don't need it anymore?

Reply:
this won't be needed in the new implementation